xcodeproj/internal/xcschemes/xcschemes.bzl

```diff
@@ -0,0 +1,1125 @@
+            [`xcschemes.pre_post_actions`](#xcschemes.pre_post_actions).
+        pre_actions: Pre-actions to run when building or running the action
+            this build target is a part of.
+
+            Elements of the `list` must be values returned by functions in
+            [`xcschemes.pre_post_actions`](#xcschemes.pre_post_actions).
+        target_environment: The
+            [target environment](#top_level_target-target_environments) to use
+            when determining which version of the top-level target
+            [`label`](#xcschemes.top_level_build_target-label) refers to.
+
+            If not set, the default target environment will be used (i.e.
+            `"simulator"` if it's one of the available target environments,
+            otherwise `"device"`).
+    """
+    if not label:
+        fail("""
+`label` must be provided to `xcschemes.top_level_build_target`.
+""")
+
+    return struct(
+        extension_host = extension_host or "",
+        include = True,
+        label = label,
+        library_targets = library_targets,
+        post_actions = post_actions,
+        pre_actions = pre_actions,
+        target_environment = target_environment,
+    )
+
+# `pre_post_actions`
+
+def _build_script(title = "Run Script", *, order = None, script_text):
+    """Defines a pre-action or post-action script to run when building.
+
+    This action will appear in the Pre-actions or Post-actions section of the
+    Build section of the scheme.
+
+    Args:
+        title: The title of the action.
+        order: The relative order of the action within the section it appears
+            in.
+
+            If `None`, the action will be added to the end of the section, in
+            an unspecified but deterministic order. Otherwise, the order should
+            be an integer. Smaller order values will run before larger order
+            values. rules_xcodeproj created actions (e.g. "Update .lldbinit and
+            copy dSYMs") use order values 0, -100, -200, etc.
+        script_text: The script text.
+
+            The script will be run in Bazel's execution root, so you probably
+            want to change to the `$SRCROOT` directory in the script.
+    """
+    if not title:
+        fail("""
+`title` must be provided to `xcschemes.pre_post_actions.build_script`.
+""")
+    if not script_text:
+        fail("""
+`script_text` must be provided to `xcschemes.pre_post_actions.build_script`.
+""")
+
+    return struct(
+        for_build = True,
+        order = order,
+        script_text = script_text,
+        title = title,
+    )
+
+def _launch_script(title = "Run Script", *, order = None, script_text):
+    """Defines a pre-action or post-action script to run when running.
+
+    This action will appear in the Pre-actions or Post-actions section of the
+    Test, Run, or Profile section of the scheme.
+
+    Args:
+        title: The title of the action.
+        order: TBD.
```

Comment:
This is still unable to be filled in, correct?

Reply:
I did finally implement it. I need to update this.